creator/ingest_runs/data_generator/study_generator.py

```diff
@@ -0,0 +1,504 @@
+"""
+A fake KF study generator that generates realistic studies
+Used to setup unit tests for ingest runs and related functionality
+
+1. (Optional) Create study files
+- Create a biological TSV file with minimal family, participant, specimen data
+- Create source and harmonized genomic files
+- Upload genomic files to study bucket on S3
+- Create S3 object manifest for source genomic files
+- Create the BIX Genomic Workflow Output Manifest
+
+2. Ingest study files into the KF Data Service
+"""
+import os
+import logging
+import random as ra
+import string
+from pprint import pprint
+
+import boto3
+from botocore.exceptions import ClientError
+import base32_crockford as b32
+import requests
+import pandas as pd
+
+from kf_lib_data_ingest.common.io import read_df
+from kf_lib_data_ingest.common import pandas_utils
+from kf_lib_data_ingest.app import settings as ingest_settings
+from kf_lib_data_ingest.etl.ingest_pipeline import DataIngestPipeline
+from d3b_utils.aws_bucket_contents import fetch_aws_bucket_obj_info
+
+# For data generation
+ROOT_DIR = os.path.abspath(os.path.dirname(os.path.dirname(__file__)))
+DATA_DIR = os.path.join(ROOT_DIR, "data_generator", "ingest_package", "data")
+INGEST_PKG_DIR = os.path.join(ROOT_DIR, "data_generator", "ingest_package")
+
+# Connections
+DATASERVICE_URL = "http://localhost:5000"
```

Comment:
We will eventually need to get this from the django app settings and not hard code it here.